internal/provider/resource_password_test.go

```diff
@@ -506,6 +506,164 @@ func TestAccResourcePassword_Min(t *testing.T) {
 	})
 }
 
+// TestAccResourcePassword_UpgradeFromVersion2_2_1 requires that you are running an amd64 Terraform binary
+// if you are running this test locally on arm64 architecture otherwise you will see the following error:
+//
+// Error: Incompatible provider version
+//
+// Provider registry.terraform.io/hashicorp/random v2.2.1 does not have a
+// package available for your current platform ...
```

Comment:
I think we can codify this logic into the test via:

```go
TestCase{
  PreCheck: func() {
    if runtime.GOOS == "darwin" && runtime.GOARCH == "arm64" {
      t.Skip("This test requires darwin/amd64 to download the old provider version. Setting TF_ACC_TERRAFORM_PATH to darwin/amd64 compatible Terraform binary can be used as a workaround.")
    }
  },
  // ... other fields
}
```

Reply:
Thanks. Have incorporated.